File: zul/src/org/zkoss/zul/api/Treefooter.java
/* Treefooter.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		Tue Oct 22 14:45:31     2008, Created by Flyworld
}}IS_NOTE

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under GPL Version 3.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
 */
package org.zkoss.zul.api;

/**
 * A column of the footer of a tree ({@link Tree}). Its parent must be
 * {@link Treefoot}.
 * 
 * <p>
 * Unlike {@link Treecol}, you could place any child in a tree footer.
 * <p>
 * Note: {@link Treecell} also accepts children.
 * <p>
 * Default {@link #getZclass}: z-tree-footer.(since 3.5.0)
 * 
 * @author tomyeh
 * @since 3.5.2
 */
public interface Treefooter extends org.zkoss.zul.impl.api.LabelImageElement {

	/**
	 * Returns the tree that this belongs to.
	 */
	public org.zkoss.zul.api.Tree getTreeApi();

	/**
	 * Returns the column index, starting from 0.
	 */
	public int getColumnIndex();

	/**
	 * Returns the tree header that is in the same column as this footer, or
	 * null if not available.
	 */
	public org.zkoss.zul.api.Treecol getTreecolApi();

	/**
	 * Returns number of columns to span this footer. Default: 1.
	 */
	public int getSpan();

	/**
	 * Sets the number of columns to span this footer.
	 * <p>
	 * It is the same as the colspan attribute of HTML TD tag.
	 */
	public void setSpan(int span);

}

File: zul/src/main/resources/web/js/zul/sel/Treefooter.ts
/* Treefooter.ts

	Purpose:

	Description:

	History:
		Wed Jun 10 15:32:42     2009, Created by jumperchen

Copyright (C) 2009 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.0 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
/**
 * A column of the footer of a tree ({@link Tree}).
 * Its parent must be {@link Treefoot}.
 *
 * <p>Unlike {@link Treecol}, you could place any child in a tree footer.
 * <p>Note: {@link Treecell} also accepts children.
 * @defaultValue {@link getZclass}: z-treefooter
 */
@zk.WrapClass('zul.sel.Treefooter')
export class Treefooter extends zul.mesh.FooterWidget {
	// NOTE: The parent of Treefooter should be Treefoot, but Treefoot is not a HeadWidget.
	// override parent!: zul.sel.Treefoot | null;

	/**
	 * @returns the tree that this belongs to.
	 */
	getTree(): zul.sel.Tree | undefined {
		return this.getMeshWidget() as zul.sel.Tree | undefined;
	}

	/**
	 * @returns the tree header that is in the same column as
	 * this footer, or null if not available.
	 */
	getTreecol(): zul.sel.Treecol | undefined {
		return this.getHeaderWidget() as zul.sel.Treecol | undefined;
	}

	/**
	 * @returns the maximal length for this cell.
	 * It is the same as the correponding {@link getTreecol}'s
	 * {@link Treecol#getMaxlength}.
	 * @since 5.0.5
	 */
	getMaxlength(): number {
		var tc = this.getTreecol();
		return tc ? tc.getMaxlength()! : 0;
	}

	/** @internal */
	override domLabel_(): string {
		return zUtl.encodeXML(this.getLabel(), {maxlength: this.getMaxlength()});
	}
}